### PR TITLE
[BIOMAGE-1803] Projects loading api v2

### PR DESCRIPTION
**Pull request by cosa65**
# Description

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1803

#### Link to staging deployment URL (or set N/A)
https://ui-martinfosco-ui673-api327.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/api/pull/327

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

**Comment by cosa65** [comment]
**tl;dr: Made mistake so I'm addressing agi's comments from [this PR](https://github.com/hms-dbmi-cellenics/ui/pull/668) in this one.**

I mixed up my PRs and ended up linking a draft PR that represented the changes in this one (and the ones in the ticket that was merged yesterday too because I was wokring on top of the other branch) to my other api PR (different tickets), so agi's review of [this PR](https://github.com/hms-dbmi-cellenics/ui/pull/668) was lost.

To address his comments I'll perform the changes from that one in this PR.

File: src/redux/actionTypes/experiments.js [hunk 1]
```diff
@@ -30,11 +30,6 @@ const EXPERIMENTS_DELETED = `${EXPERIMENTS}/deleted`;
  */
 const EXPERIMENTS_SAVING = `${EXPERIMENTS}/saving`;
 
-/**
- * Experiment saved
- */
-const EXPERIMENTS_SAVED = `${EXPERIMENTS}/saved`;
-
```
Comment (ivababukova) [inline src/redux/actionTypes/experiments.js:37]: why is this removed?

Comment (ivababukova) [reply src/redux/actionTypes/experiments.js:37]: nevermind, I see why -- because the `saveExperiment` action is being removed

File: src/redux/reducers/experiments/experimentsSaved.js [hunk 2]
```diff
@@ -1,9 +0,0 @@
-const experimentSaved = (state) => ({
-  ...state,
-  meta: {
-    ...state.meta,
-    saving: false,
-  },
-});
-
```
Comment (ivababukova) [inline src/redux/reducers/experiments/experimentsSaved.js:8]: you sure that we can remove this? And there is no need to change the redux store after and experiment is updated? Or are the changes of the store happening from other reducers?

File: src/redux/actions/projects/loadProjects.js [hunk 3]
```diff
@@ -1,29 +1,83 @@
...
+  const projectsListV1 = experimentListV2.map((experimentData) => {
+    const {
+      createdAt,
+      updatedAt,
+      id,
+      samplesOrder,
+      ...restOfExperimentData
+    } = experimentData;
+
+    // Not assigning "lastAnalyzed" because that is no longer used anywhere
```
Comment (ivababukova) [inline src/redux/actions/projects/loadProjects.js:22]: isn't lastAnalysed used in data management to display the dates for `Modified` and `Created` for projects?
Screenshot 2022-04-05 at 09 59 40

File: src/redux/actions/projects/updateProject.js [hunk 4]
```diff
@@ -25,10 +25,9 @@ const updateProject = (
   const newProject = mergeObjectWithArrays(currentProject, diff);
 
   try {
+    // if config.currentApiVersion === api.V2 dont do any fetch, updating the experiment is enough
```
Comment (ivababukova) [inline src/redux/actions/projects/updateProject.js:28]: do you mean that updating the project is enough? I see that we dispatch `PROJECTS_UPDATE` action type on line 33

Comment (ivababukova) [reply src/redux/actions/projects/updateProject.js:28]: why did you comment out the if statement as well? It looks more clear to me if the if statement is in code

File: src/redux/reducers/experiments/experimentsSaved.js (hunk 2 above)
Comment (ivababukova) [reply src/redux/reducers/experiments/experimentsSaved.js:8]: nevermind, I see that you moved this to the `experimentsUpdate` reducer

File: src/redux/actions/projects/loadProjects.js (hunk 3 above)
Comment (cosa65) [reply src/redux/actions/projects/loadProjects.js:22]: No, these are `createdDate` and `lastModified` (`createdAt` and `updatedAt` in api v2)

File: src/redux/actions/projects/updateProject.js (hunk 4 above)
Comment (cosa65) [reply src/redux/actions/projects/updateProject.js:28]: > why did you comment out the if statement as well? It looks more clear to me if the if statement is in code

@aerlaut commented to remove the if https://github.com/hms-dbmi-cellenics/ui/pull/668#discussion_r837309100 :P

Personally I don't mind either way so tell me what to do when there's an agreement.


> do you mean that updating the project is enough? I see that we dispatch PROJECTS_UPDATE action type on line 33

I mean that updating the experiment in the api is enough, not in redux

Comment (ivababukova) [reply src/redux/actions/projects/updateProject.js:28]: about the commented out if statement: it is up to you, it seems to be a matter of personal preference. Happy with either option, not a big deal.

Comment (ivababukova) [reply src/redux/actions/projects/updateProject.js:28]: `I mean that updating the experiment in the api is enough, not in redux` <--- sounds good

**Review by ivababukova — APPROVED** [review]
